### PR TITLE
Remove unused permissions; add missing watch; Makefile improvements

### PR DESCRIPTION
**Pull request by grdryn**
JIRA: https://issues.jboss.org/browse/AEROGEAR-9616

## Verification steps

1.  Check out this branch locally
1. `make cluster/clean cluster/prepare install-operator install-mdc`
1. Make sure there aren't any errors in the operator logs that could be caused by restricted permissions
1. Make sure there are no errors in the mdc logs that could be caused by restricted permissions
1. Do some things in MDC, like create an app & bind something, & make sure they work as expected. I don't know how to use MDC :see_no_evil:

### CODE REVIEW

**Comment by aliok** [comment]
:eye:

File: deploy/role_binding.yaml [hunk 1]
```diff
@@ -9,3 +10,17 @@ roleRef:
...
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: mobileclient-admin-operator
+subjects:
+- kind: ServiceAccount
+  name: mobile-developer-console-operator
+roleRef:
+  kind: Role
+  name: mobileclient-admin
```
Comment (aliok) [inline deploy/role_binding.yaml:25]: This role binding makes perfect sense. Thanks for code deduplication.

File: pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole_controller.go [hunk 2]
```diff
@@ -52,26 +52,26 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	// Watch for changes to secondary resource DeploymentConfig and requeue the owner MobileDeveloperConsole
```
Comment (aliok) [inline pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole_controller.go:55]: Any particular reason for the reordering here?

**Comment by aliok** [comment]
Verified (didn't try every single CR). Code also looks good.
Thanks for the makefile improvements as well!

Feel free to merge.

File: pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole_controller.go (hunk 2 above)
Comment (grdryn) [reply pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole_controller.go:55]: The watch for `Service` was missing, so I just sort of re-ordered them all based on their groups while I was adding that: the Kubernetes core ones are first, then the other k8s groups, then the OpenShift ones are at the bottom.

**Comment by grdryn** [comment]
Thanks a lot for the review and verification! :smiley:

File: pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole_controller.go (hunk 2 above)
Comment (aliok) [reply pkg/controller/mobiledeveloperconsole/mobiledeveloperconsole_controller.go:55]: ok, sounds great. feel free to merge

**Pull request merged**